test-loop-tests/src/tests/indexer.rs

```diff
@@ -192,20 +211,73 @@ fn shutdown(env: TestLoopEnv) {
     env.shutdown_and_drain_remaining_events(Duration::seconds(20));
 }
 
+fn next_nonce() -> Nonce {
+    static NEXT_VALUE: AtomicU64 = AtomicU64::new(1);
+    NEXT_VALUE.fetch_add(1, std::sync::atomic::Ordering::Relaxed)
+}
+
+fn tx_block_hash(env: &TestLoopEnv) -> CryptoHash {
+    TestLoopNode::rpc(&env.node_datas).head(env.test_loop_data()).last_block_hash
+}
+
+fn user_signer() -> Signer {
+    create_user_test_signer(&user_account())
+}
+
+fn send_validator_tx(env: &TestLoopEnv, tx: SignedTransaction) {
+    let process_tx_request =
+        ProcessTxRequest { transaction: tx, is_forwarded: false, check_only: false };
+    TestLoopNode::validator(&env.node_datas, 0).data().rpc_handler_sender.send(process_tx_request);
+}
+
 fn create_local_tx(env: &TestLoopEnv) -> SignedTransaction {
-    let rpc_node = TestLoopNode::rpc(&env.node_datas);
-    let block_hash = rpc_node.head(env.test_loop_data()).last_block_hash;
-    let nonce = 1;
-    let tx = SignedTransaction::call(
-        nonce,
+    SignedTransaction::call(
+        next_nonce(),
         user_account(),
         user_account(),
-        &create_user_test_signer(&user_account()),
+        &user_signer(),
         Balance::ZERO,
         "does_not_matter".to_owned(),
         vec![],
         Gas::from_teragas(100),
-        block_hash,
+        tx_block_hash(env),
+    )
+}
+
+fn create_burn_gas_tx(env: &TestLoopEnv) -> SignedTransaction {
+    let gas_to_burn = GAS_LIMIT.checked_div(2).unwrap().checked_add(Gas::from_gas(1)).unwrap();
```

Comment:
nit: It would be easier to see what is happening with either gas_to_burn being an argument to create_burn_gas_tx or create_burn_gas_tx being inlined.